clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3646,23 +3630,167 @@ static void handleSYCLIntelSchedulerTargetFmaxMhzAttr(Sema &S, Decl *D,
 }
 
 // Handles max_global_work_dim.
-static void handleMaxGlobalWorkDimAttr(Sema &S, Decl *D, const ParsedAttr &A) {
-  if (D->isInvalidDecl())
-    return;
+// Returns a ArgCheck value; EqualToOne means all argument values are
+// equal to one, NotEqualToOne means all argument values are not
+// equal to one, and Unknown means that the argument values are invalid.
+enum class ArgCheck { Unknown, EqualToOne, NotEqualToOne };
```

Comment:
I have changed ``ArgCheck`` to to ``OneArgResult`` since we are checking if the all arguments are equal to one here. 
I am open for any other name.  Thanks @AaronBallman